core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -3472,6 +3483,300 @@ protected UniqueIDPaginatedSearchResult doGetUserListWithID(Condition condition,
         return result;
     }
 
+    @Override
+    public String doGetGroupIdFromGroupName(String groupName) throws UserStoreException {
+
+        String sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.GET_GROUP_ID_FROM_GROUP_NAME);
+
+        try (Connection dbConnection = getDBConnection();
+             PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt)) {
+
+            prepStmt.setString(1, groupName);
+            if (sqlStmt.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
+                prepStmt.setInt(2, tenantId);
+            }
+            try (ResultSet resultSet =  prepStmt.executeQuery()) {
+                if (resultSet.next()) {
+                    return resultSet.getString(1);
+                }
+                return null;
+            }
+        } catch (SQLException e) {
+            String msg = "Error occurred while retrieving group id for group : " + groupName;
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new UserStoreException(msg, e);
+        }
+    }
+
+    @Override
+    public String doGetGroupNameFromGroupId(String groupId) throws UserStoreException {
+
+        String groupName = "";
+        String sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.GET_GROUP_NAME_FROM_GROUP_ID);
+
+        try (Connection dbConnection = getDBConnection();
+             PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt)) {
+            prepStmt.setString(1, groupId);
+            if (sqlStmt.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
+                prepStmt.setInt(2, tenantId);
+            }
+            try (ResultSet resultSet =  prepStmt.executeQuery()) {
+                if (resultSet.next()) {
+                    groupName = resultSet.getString(1);
+                }
+            }
+        } catch (SQLException e) {
+            String msg = "Error occurred while retrieving group name for group with ID: " + groupId;
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new UserStoreException(msg, e);
+        }
+        String groupNameWithDomain = UserCoreUtil.addDomainToName(groupName, getMyDomainName());
```

Comment:
We do not need this variable. Also, what will happen when the group name is empty?

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/3847/commits/344924abcc35406833451b3a6c68ab68ec225912